pkgs/servers/zoneminder/default.nix

```diff
@@ -182,6 +182,9 @@ in stdenv.mkDerivation rec {
     ln -s $out/share/zoneminder/www $out/share/zoneminder/www/zm
   '';
 
+  # libdl.so.2: error adding symbols: DSO missing from command line
+  LDFLAGS = "-ldl";
```

Comment:
What triggers this error?

Reply:
I don't know. Maybe it's the order of the linking flags. That's part of my output:
```
[100%] Linking CXX executable zma                                                                                                                                                                                                              
/nix/store/qw8rhqk004qmqgnkvkpj85l1s194ang1-binutils-2.31.1/bin/ld: /nix/store/xm8fav5cy5kmjqzqj1l7yqhnikj4i6ff-mariadb-10.3.15-dev/lib/libmysqlclient.a(ma_client_plugin.c.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'           
/nix/store/qw8rhqk004qmqgnkvkpj85l1s194ang1-binutils-2.31.1/bin/ld: /nix/store/avr2x43njlq4kyb1a9zgrh6fih5fq4si-glibc-2.27/lib/libdl.so.2: error adding symbols: DSO missing from command line                                                 
collect2: error: ld returned 1 exit status                                                                                                                                                                                                     
make[2]: *** [src/CMakeFiles/zma.dir/build.make:101: src/zma] Error 1                                                                                                                                                                          
make[1]: *** [CMakeFiles/Makefile2:269: src/CMakeFiles/zma.dir/all] Error 2
```

I hope I fixed it in the correct way, I don't know.

Reply:
I'm curious what changed and broke the build though.

Reply:
grep says that `bin/{zma,zmc,zmu}` all contain `dlclose@@GLIBC_2.2.5`

Reply:
Maybe upstream has an idea ZoneMinder/zoneminder#2648

Reply:
I fixed it upstream and am now using that patch. Then we'll get error when upgrading and know to remove the patch. Otherwise we might forget to remove the `LDFLAGS`.